custom_code/forms.py

```diff
@@ -2,7 +2,276 @@
 from tom_targets.models import TargetList
 from .models import TargetListExtra
+from datetime import datetime
+import json
+import os
 
 
 TargetListExtraFormset = inlineformset_factory(TargetList, TargetListExtra, fields=('key', 'value'),
                                                widgets={'value': forms.TextInput()})
+
+TNS_FILTER_CHOICES = [
+    (0, "Other"),
+    (1, "Clear"),
+    (10, "U-Johnson"),
+    (11, "B-Johnson"),
+    (12, "V-Johnson"),
+    (13, "R-Cousins"),
+    (14, "I-Cousins"),
+    (15, "J-Bessel"),
+    (16, "H-Bessel"),
+    (17, "K-Bessel"),
+    (18, "L"),
+    (19, "M"),
+    (20, "u-Sloan"),
+    (21, "g-Sloan"),
+    (22, "r-Sloan"),
+    (23, "i-Sloan"),
+    (24, "z-Sloan"),
+    (25, "y-P1"),
+    (26, "w-P1"),
+]
+
+TNS_INSTRUMENT_CHOICES = [
+    (0, "Other"),
+]
+
+TNS_CLASSIFICATION_CHOICES = [
+    (0, "Other"),
+    (1, "SN"),
+    (2, "SN I"),
+    (3, "SN Ia"),
+    (4, "SN Ib"),
+    (5, "SN Ic"),
+    (6, "SN Ib/c"),
+    (7, "SN Ic-BL"),
+    (9, "SN Ibn"),
+    (10, "SN II"),
+    (11, "SN IIP"),
+    (12, "SN IIL"),
+    (13, "SN IIn"),
+    (14, "SN IIb"),
+    (15, "SN I-faint"),
+    (16, "SN I-rapid"),
+    (18, "SLSN-I"),
+    (19, "SLSN-II"),
+    (20, "SLSN-R"),
+    (23, "Afterglow"),
+    (24, "LBV"),
+    (25, "ILRT"),
+    (26, "Nova"),
+    (27, "CV"),
+    (28, "Varstar"),
+    (29, "AGN"),
+    (30, "Galaxy"),
+    (31, "QSO"),
+    (40, "Light-Echo"),
+    (50, "Std-spec"),
+    (60, "Gap"),
+    (61, "Gap I"),
+    (62, "Gap II"),
+    (65, "LRN"),
+    (66, "FBOT"),
+    (70, "Kilonova"),
+    (99, "Impostor-SN"),
+    (100, "SN Ia-pec"),
+    (102, "SN Ia-SC"),
+    (103, "SN Ia-91bg-like"),
+    (104, "SN Ia-91T-like"),
+    (105, "SN Iax[02cx-like]"),
+    (106, "SN Ia-CSM"),
+    (107, "SN Ib-pec"),
+    (108, "SN Ic-pec"),
+    (109, "SN Icn"),
+    (110, "SN Ibn/Icn"),
+    (111, "SN II-pec"),
+    (112, "SN IIn-pec"),
+    (115, "SN Ib-Ca-rich"),
+    (116, "SN Ib/c-Ca-rich"),
+    (117, "SN Ic-Ca-rich"),
+    (118, "SN Ia-Ca-rich"),
+    (120, "TDE"),
+    (121, "TDE-H"),
+    (122, "TDE-He"),
+    (123, "TDE-H-He"),
+    (200, "WR"),
+    (201, "WR-WN"),
+    (202, "WR-RC"),
+    (203, "WR-WO"),
+    (210, "M dwarf"),
+]
+
+class TargetReportForm(forms.Form):
+    ra = forms.FloatField()
+    dec = forms.FloatField()
+    reporting_group_id = forms.ChoiceField(choices=[
+        (66, "SAGUARO"),
+    ], initial=(66, "SAGUARO"))
+    discovery_data_source_id = forms.ChoiceField(choices=[
+        (66, "SAGUARO"),
+    ], initial=(66, "SAGUARO"))
+    reporter = forms.CharField()
+    discovery_datetime = forms.DateTimeField(initial=datetime.utcnow())
+    at_type = forms.ChoiceField(choices=[
+        (0, "Other"),
+        (1, "PSN"),
+        (2, "PNV"),
+        (3, "AGN"),
+        (4, "NUC"),
+        (5, "FRB"),
+    ], initial=(1, "PSN"))
+    non_detection__archiveid = forms.ChoiceField(choices=[
+        (0, "Other"),
+        (1, "SDSS"),
+        (2, "DSS"),
```

Comment:
Should this be CSS?

Reply:
It actually is DSS (Digitized Sky Survey)! We may eventually want to ask them to add more options, but for now I just included what the TNS currently has.